src/components/Select/Select.tsx

```diff
@@ -25,7 +26,12 @@ import Text from '../Text';
 
 import styles from './Select.module.css';
 
-type SelectProps = ExtractProps<typeof Listbox> &
+// https://github.com/tailwindlabs/headlessui/blob/%40headlessui/react%40v1.7.19/packages/%40headlessui-react/src/components/listbox/listbox.tsx#L349
+type SelectProps = ListboxProps<
+  ElementType,
+  string | { [k: string]: unknown },
+  { [k: string]: unknown }
```

Comment:
the first line represents what is allowed in things like `defaultValue` or `value`

the second line represents what gets used when `by` is evaluated

* when a simple string, `by` is string
* when an object, `by` is the type of the keys on that object (usually `string | number`), or that eval function
* ...